backends/bmv2/midend.cpp

```diff
@@ -122,6 +123,11 @@ MidEnd::MidEnd(BMV2Options& options) {
         new P4::CompileTimeOperations(),
         new P4::TableHit(&refMap, &typeMap),
         new P4::RemoveLeftSlices(&refMap, &typeMap),
+
+        // p4c-bm removed unused action parameters. To produce a compatible
+        // control plane API, we remove them as well for P4-14 programs.
+        isv1 ? new P4::RemoveUnusedActionParameters(&refMap) : nullptr,
```

Comment:
I believe the answer is no, because in that case we were just skipping them when writing the JSON output, but in this case we're actually cleanly removing them from the IR at an earlier stage.